airflow/providers/databricks/operators/databricks.py

```diff
@@ -1041,7 +1044,21 @@ def monitor_databricks_job(self) -> None:
         run = self._hook.get_run(self.databricks_run_id)
         run_state = RunState(**run["state"])
         self.log.info("Current state of the job: %s", run_state.life_cycle_state)
+
         while not run_state.is_terminal:
+            if self.deferrable:
```

Comment:
this looks a bit wrong, this should happen before the while loop

Reply:
I will refactor this. 

But, the reason for doing this is if the Databricks job exited even before we deferred it, I thought this would be optimal. Although, I agree that this is a rare case scenario and better to refactor

Reply:
We can add a conditional check `if not run_state.is_terminal:`  before deferring to check if the job still is running. And the while loop can follow it later.

Reply:
Updated it